src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -177,20 +189,29 @@ public override void Return(T[] array, bool clearArray = false)
                 // if there was a previous one there, push that to the global stack.  This
                 // helps to keep LIFO access such that the most recently pushed stack will
                 // be in TLS and the first to be popped next.
-                T[][] tlsBuckets = t_tlsBuckets;
+                Bucket[] tlsBuckets = t_tlsBuckets;
                 if (tlsBuckets == null)
                 {
-                    t_tlsBuckets = tlsBuckets = new T[NumBuckets][];
-                    tlsBuckets[bucketIndex] = array;
+                    t_tlsBuckets = tlsBuckets = new Bucket[NumBuckets];
+                    tlsBuckets[bucketIndex].Buffer = array;
```

Comment:
Seems to me you want to set the TIcks field  here as well (you are putting something in the bucket, you want to know the time you do it.   This is the rare path anyway, so we should not feel bad about it.

Reply:
Adding